docs/spfx/web-parts/guidance/getting-started-with-top-actions.md

```diff
@@ -0,0 +1,170 @@
+
+## Code Snippets
+### Button command
+The type interace for a button is similar to the property panel's button (`IPropertyPaneButtonProps`).
+
+```typescript
+import { ITopActions } from '@microsoft/sp-top-actions';
+import { PropertyPaneFieldType } from '@microsoft/sp-property-pane';
+...
+public getTopActionsConfiguration(): ITopActions | undefined {
+  return {
+    topActions: [
+      {
+        targetProperty: 'reset',
+        type: PropertyPaneFieldType.Button,
+        properties: {
+          icon: 'Reset'
+        }
+      }
+    ],
+    onExecute: (actionName: string, newValue: any) => {
+      if (actionName === 'reset') {
+        // user defined logic to reset the web part
+        this.reset();
+      }
+    }
+  };
+}
+```
+
+### Drop down command
+The type interace for a drop down is similar to the property panel's choice group (`IPropertyPaneChoiceGroupOption`).
+
+```typescript
+import { ITopActions } from '@microsoft/sp-top-actions';
+import { PropertyPaneFieldType } from '@microsoft/sp-property-pane';
+...
+public getTopActionsConfiguration(): ITopActions | undefined {
+  return {
+    topActions: [{
+      targetProperty: 'layout',
+      type: PropertyPaneFieldType.ChoiceGroup,
+      properties: {
+        options: [
+          {
+            // key maps to newValue in onExecute
+            key: 'card',
+            text: 'Card Layout',
+            imageSize: { width: 32, height: 32 },
+            iconProps: { officeFabricIconFontName: 'ArticlesIcon' },
+            checked: this.state.layout === 'card'
+          },
+          {
+            key: 'list',
+            text: 'List Layout',
+            imageSize: { width: 32, height: 32 },
+            // you can use iconProps, icon to define icons
+            icon: 'List',
+            checked: this.state.alignment === 'list'
+          }
+        ]
+      }
+    }],
+    // for ChoiceGroup drop down, the newValue tells us which option's key was selected
+    onExecute: (actionName: string, newValue: any) => {
+      if (actionName === 'layout') {
+        this.setLayout(newValue);
+        this.render();
+      }
+    }
+  };
+}
+```
+
+## Advanced configurations
+For advanced configurations of your top action commands, checkout the type definitions from `@microsoft/sp-property-pane` and `@microsoft/sp-top-actions`. Currently, the two supported top action commands, button and drop down, can be defined using a subset of the types `IPropertyPaneChoiceGroupOption` and `IPropertyPaneButtonProps`.
+
+For `IPropertyPaneButtonProps`, the currently supported properties are `icon`, `text`, `ariaLabel`, `disabled`
+
+For `IPropertyPaneChoiceGroupOption`, the currently supported porperty is `options` and from that array we support `key`, `text`, `iconProps.officeFabricIconFontName`, `imageSize`, `checked`, `title`
+
+> [!WARNING]

```

Comment:
This isn't necessary... if the feature is in preview/beta, use the provided includes at the top of the doc to indicate it this is a preview/beta feature. This makes it easier for us to find instances of things marked as "beta/preview" in the docs in a release to remove when something goes GA.

See this for reference: https://github.com/SharePoint/sp-dev-docs/compare/main...andrewconnell:sp-dev-docs:content-update/spfx-v1.16#diff-5563a3a364d59466a3d5a799f86d4318e6a9c6e3c309b4932de7f434e23a2332R9